app/goal_routes.py

```diff
@@ -0,0 +1,106 @@
+from flask import Blueprint, request, make_response, jsonify, abort
+from app import db
+from app.models.goal import Goal
+from app.task_routes import validate_task_id
+
+def validate_goal_id(goal_id):
+    try: 
+        goal_id = int(goal_id)
+    except ValueError:
+        abort(make_response({"msg":f"The goal with id {goal_id} is not found"}, 400))
```

Comment:
It's a small thing but this error message is incorrect:

```suggestion
        abort(make_response({"msg":f"Invalid goal id \"{goal_id}\""}, 400))
```